firewalls.tf

```diff
@@ -1,8 +1,8 @@
 resource "google_compute_firewall" "allow_metastore_egress" {
   count       = var.enable_dataproc_network ? 1 : 0
-  project     = google_compute_network.vpc_network[*].project
+  project     = google_compute_network.vpc_network[0].project
```

Comment:
why is this 0 instead of *? 

I think I know why but can we confirm this doesn't affect anything in the plan? If so we should just change it back (unless you or someone in the team feels strongly about this)

Reply:
I had plan errors with `*` 

```
│ Error: Incorrect attribute value type
│ 
│   on .terraform/modules/portrait_engine_data_planes/firewalls.tf line 27, in resource "google_compute_firewall" "allow_idapi_egress":
│   27:   network     = google_compute_network.vpc_network[*].name
│     ├────────────────
│     │ google_compute_network.vpc_network is tuple with 1 element
```

Reply:
semantically, I feel it should be a '*' because google_compute_network.vpc_network returns an array. So, my vote here is to not touch and leave as it was. 

Regarding the error you got @KDodhi very weird, might need investigating to understand, as this original value already existed before and the module is in use.